Add page level layout components

yoshiemuranaka opened the pull request:
Adding components to help standardize type, size, spacing
for common elements on a page.

Closes #55

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Add page level layout components

Adding components to help standardize type, size, spacing
for common elements on a page.

Closes #55

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Merge branch 'master' into page-template

derick-montague commented on src/components/Global/PageContainer.vue:
```diff
@@ -0,0 +1,18 @@
+<style lang="scss" scoped>
+main {
+  margin-top: 24px;
+  margin-bottom: 50px;
```
Wondering if we should use the `$spacer` token.

```
  margin-top: $spacer * 1.5;
  margin-bottom: $spacer * 3;
```

derick-montague replied:
Wondering if we should use the `$spacer` token

```
  margin-top: $spacer * 1.5;
  margin-bottom: $spacer * 3;
```

derick-montague commented on src/components/Global/PageSection.vue:
```diff
@@ -0,0 +1,30 @@
+
+<style lang="scss" scoped>
+.page-section {
+  margin-bottom: 2rem;
```
` margin-bottom: $spacer * 2;`

derick-montague commented on src/components/Global/PageSection.vue:
```diff
@@ -0,0 +1,30 @@
+}
+h2 {
+  @include font-size($h4-font-size);
+  margin-bottom: 1rem;
```
`margin-bottom:  $spacer`

derick-montague commented on src/components/Global/PageTitle.vue:
```diff
@@ -0,0 +1,26 @@
+  margin-bottom: 2rem;
+}
+.page-description {
+  max-width: 750px;
```
Is this to handle character spacing limits? I think this will work, but wonder about future scaling. One option, although not perfect either is to set the width between 60ch and 72ch.

derick-montague commented on src/components/Global/PageTitle.vue:
```diff
@@ -0,0 +1,26 @@
+  line-height: $display-line-height;
+}
+.page-title {
+  margin-bottom: 2rem;
```
`margin-bottom: $spacer * 2;`

derick-montague reviewed (commented):
A few small suggestions to consider. Thanks!

yoshiemuranaka committed:
Update page template components

- Use $spacer tokens for margin
- Update max width px to ch

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

yoshiemuranaka committed:
Merge branch 'master' into page-template

yoshiemuranaka committed:
Fix linting issues

Updating new vue files to use single quotes instead of
double.
Also removed a conflicting classname and added modifications
to the main container. The main container needed a min-height
and height value set, which became apparent with added
background color. Adding a background color will move us
closer to agreed design solution to add a sublte background
color instead of adding a border to separate main content from
left hand navigation.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented on src/views/AccessControl/LocalUserManagement/LocalUserManagement.vue:
```diff
@@ -1,10 +1,6 @@
-        <h1>Local user management</h1>
-      </b-col>
-    </b-row>
+    <PageTitle pageTitle="Local user management" />
```
How do you feel about pulling this from the router meta data? 
You could bind the route meta.title...
`<PageTitle :pageTitle="this.$route.meta.title" />`

yoshiemuranaka replied:
Yeah, I think that's a good idea, I'll just remove the title prop all together and always pull from route meta

derick-montague commented on src/components/Global/PageTitle.vue:
```diff
@@ -0,0 +1,26 @@
+<script>
+export default {
+  name: 'PageTitle',
+  props: ['pageTitle', 'pageDescription']
```
Also wondering if we need to be this verbose in our prop naming? Is there a reason we wouldn't use the prop name `title` opposed to `propTitle`? Same with description. This is not a big deal, just curious

yoshiemuranaka replied:
yeah, agree, will plan to remove title prop all together and change to description

yoshiemuranaka committed:
Use route meta data for page title

Remove title prop from PageTitle component and just
pull from route meta data. Renamed route titles to
match the navigation.
Also renamed pageDescription prop to be more succinct.

Signed-off-by: Yoshie Muranaka <yoshiemuranaka@gmail.com>

derick-montague commented:
Migrated to webui-vue in https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28730
